src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        GeometricFigure geometricFigure = new GeometricFigureSupplier().getRandomFigure();
+        int random = new Random().nextInt(1, 100);
```

Comment:
The instantiation of 'Random' should be done once and reused, instead of creating a new instance every time the method is called.